### PR TITLE
Expand mypy config, following HomeAssistant

### PR DESCRIPTION
**Pull request by alexrudd2**
Builds on #1388

See #1360

### CODE REVIEW

**Comment by janiversen** [comment]
Looks good. Hope the most parts can be done without tooo much editing.

**Comment by alexrudd2** [comment]
> Looks good. Hope the most parts can be done without tooo much editing.

Enabling all the rules gives ~1000 errors, mostly `disallow_untyped_defs`.  I think we should more cleanly define internal versus external methods, since external ones are more important to be typed correctly.

**Comment by janiversen** [comment]
That is a plan I highly agree with. As I work through the code, all real externals will be or are defined in __init__ as external symbols, that can at least to identify the external interfaces.

Can e.g. "disallow_untyped_defs" be restricted to a module ? if so, we can consider making a API module, and have all external classes (pure interface classes there). That does not sound difficult to do, and will just affect the import statement in the apps.

**Comment by janiversen** [comment]
Once you update your branch (also in the other PRs) you will possibly see more problems.

**Comment by alexrudd2** [comment]
> Can e.g. "disallow_untyped_defs" be restricted to a module ? if so, we can consider making a API module, and have all external classes (pure interface classes there). That does not sound difficult to do, and will just affect the import statement in the apps.

I believe so, yes.

**Comment by janiversen** [comment]
and you have a couple of conflicts to solve...

**Comment by alexrudd2** [comment]
> and you have a couple of conflicts to solve...

Yes, I am rebasing now locally

File: pymodbus/server/async_io.py [hunk 1]
```diff
@@ -873,7 +874,7 @@ class ModbusSerialServer:  # pylint: disable=too-many-instance-attributes
     server context instance.
     """
 
-    handler = None
+    handler: ModbusSingleRequestHandler = None
```
Comment (janiversen) [inline pymodbus/server/async_io.py:877]: Handler can be all 3 handlers, but I am ok with only typing one of them.

**Review by janiversen — APPROVED** [review]
LGTM.

**Pull request merged**